src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.constructor.alive.push(this);
+  }
+
+  static removeAnimal() {
+    this.alive = this.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Carnivore) && !animal.hidden) {
```

Comment:
I'd advise to handle damage from bite inside `Herbivore` class. Health reduction by 50 is the feature of herbivores, so it's better to define such feature inside their class. Here's how it can be implemented:
```
class Herbivore extends Animal {
  ...
  receiveBite() {
    if (!this.hidden) {
      this.health -= 50;
      if (this.health <= 0) {
        Animal.alive = ...
      }
    }
  }
}

class Carnivore extends Animal {
  ...
  bite(animal) {
    if (animal instanceof Herbivore) {
      animal.receiveBite();
    }
  }
}
```
This would make classes design clearer and easier to maintain. For example: if herbivore would have a different effect when it's bitten, it'd be easier to change that effect in `Herbivore` class rather than in `Carnivore`.